### PR TITLE
Math::FitsInDWord bugfix

### PR DESCRIPTION
**Pull request by Cellule**
Fix Math::FitsInDWord(int64) to account for negative value. Also, values like 0x00F0000000 were invalid because (0x00F0000000 & 0xFFFFFFFF) == 0xF0000000 (int64)0xF0000000 == 0xFFFFFFFFF0000000

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/2371)

### CODE REVIEW

**Comment by Cellule** [comment]
@LouisLaf @MikeHolman please review.

**Comment by Cellule** [comment]
@Microsoft/chakra-developers could someone take a look at this please ?

**Comment by suwc** [comment]
@Cellule shall we add some tests?

**Comment by Cellule** [comment]
@suwc The repro I had was pretty big, I'll see if I can reduce it to make it a test

File: lib/Common/Common/MathUtil.h [hunk 1]
```diff
@@ -44,17 +44,8 @@ class Math
...
-    {
-#if TARGET_32
-        CompileAssert(sizeof(size_t) == 4);
-        return FitsInDWord((int32) value);
-#elif TARGET_64
-        CompileAssert(sizeof(size_t) == 8);
-        return FitsInDWord((int64)value);
-#endif
-    }
+    static bool     FitsInDWord(size_t value) { return ((size_t)(signed int)(value & 0xFFFFFFFF) == value); }
```
Comment (obastemur) [inline lib/Common/Common/MathUtil.h]: Curious; Why don't you compare `value` to `INT_MIN` and `INT_MAX` range instead of forcing to overflow (signed int)?

Comment (Cellule) [inline lib/Common/Common/MathUtil.h]: This was the original implementation, I believe it requires fewer instructions this way than doing 2 signed comparisons.

File: lib/Common/Common/MathUtil.h [hunk 2]
```diff
@@ -44,17 +44,8 @@ class Math
...
+    static bool     FitsInDWord(int64 value) { return FitsInDWord((size_t)value); }
```
Comment (LouisLaf) [inline lib/Common/Common/MathUtil.h]: >static bool     FitsInDWord(int64 value) { return FitsInDWord((size_t)value); } [](start = 4, length = 79)

This doesn't make much sense on 32bits...

Comment (Cellule) [inline lib/Common/Common/MathUtil.h]: humm right...
Should I reimplement it like the size_t version aka
`return ((int64)(signed int)(value & 0xFFFFFFFF) == value);`?

Comment (LouisLaf) [inline lib/Common/Common/MathUtil.h]: Yes, that's what I'd do.

**Comment by LouisLaf** [comment]
:shipit:

**Pull request merged**